### ISSUE TITLE
Missing data columns where not handled correctly (<= 0.3)

### ISSUE THREAD

**Issue by edwindj** [data-cleaning/errorlocate#22]
If we supply a dataset missing a column/variable which is mentioned in the rules set. It just ignored the variable or gave strange errors.

``` r
library(errorlocate)
rules <- validator(x > 0, y > 0)
# loading data without y!
data <- data.frame(x = 1)
le <- locate_errors(data, rules)
le$errors
```

    ##          x
    ## [1,] FALSE

``` r
rules <- validator(x > y, y > 0, z > 0)
data <- data.frame(x = 0, y = 1)
le <- locate_errors(data, rules)
le$errors
```

    ##         x  y
    ## [1,] TRUE NA

**Comment by edwindj** [comment]
Fixed in version 0.3.2, 
Automatically a column is added to the dataset with NA and a warning is generated.


``` r
library(errorlocate)
rules <- validator(x > 0, y > 0)
data <- data.frame(x = 1)
le <- locate_errors(data, rules)
```

    ## Warning: Adding missing columns 'y'=NA to data.frame.

``` r
le$errors
```

    ##          x  y
    ## [1,] FALSE NA

``` r
rules <- validator(x > y, y > 0, z > 0)
data <- data.frame(x = 0, y = 1)
le <- locate_errors(data, rules)
```

    ## Warning: Adding missing columns 'z'=NA to data.frame.

``` r
le$errors
```

    ##         x     y  z
    ## [1,] TRUE FALSE NA

**Comment by edwindj** [comment]
Choose a different solution: missing a data column now gives an error. (>=0.3.3)
